### ISSUE TITLE
Support a date time in intermediate, non-/interrupting timer boundary / event  subprocesses

### ISSUE THREAD

**Issue by asaf** [camunda/zeebe#5141]
[datetime.bpmn.zip](https://github.com/zeebe-io/zeebe/files/5050042/datetime.bpmn.zip)

Time Date should work for a specific point in time:

Current Results:

```
value: 2019-10-01T12:00:00Z
error: Expected result of the expression '2019-10-01T12:00:00Z' to be parsed to duration, but was '2019-10-01T12:00:00Z' and encountered error: Text cannot be parsed to a Period


value: = date and time("2019-10-01T12:00:00Z")
error: Expected result of the expression ' date and time("2019-10-01T12:00:00Z")' to be one of '[DURATION, PERIOD, STRING]', but was 'DATE_TIME'
```

The only workaround I found is to use two dates to calc duration such:

```
= date and time("2019-10-01T12:00:00Z")- date and time(now)
```

Related to #3655 and #5140 

Thanks.

**Comment by saig0** [comment]
@asaf, thank you for raising this up. 

Currently, a date timer is supported for timer start events only. I assume that you want to use date timers also for
* intermediate timer catch events
* non-/interrupting timer boundary events
* non-/interrupting timer event subprocesses

Is this correct?

**Comment by asaf** [comment]
@saig0 Exactly

**Comment by themichaelscott** [comment]
I'm surprised that this doesn't work. I did this in May/June 2020 and it worked perfectly with a date object.  Now, suddenly when I tried to use it again, it no longer works.  What is the ETA on this to be able to accept a date/time like 2019-10-01T12:00:00Z on an intermediate timer catch event?

**Comment by mumukiller** [comment]
Guys could you please confirm that the issue related to zeebe engine itself, not to zeebe modeler? I'm asking because there is no chance to select date time in zeebe modeler either

**Comment by saig0** [comment]
Currently, date-time definitions are **officially** only supported for timer start events. However, date-time definitions may work also for other timer events. I just tested it with an intermediate timer event and it works. Internally, it is using the same code. But other events than timer start events are not covered by our tests. 
In order to support the date-time definition for all timer events, we would need to write more test cases and extend the modeler.

**Comment by mumukiller** [comment]
@saig0 could you please share a piece of bpmn file from your example of how date-time can be used?

**Comment by saig0** [comment]
Just replace the timer definition of a timer event in the XML. A snippet of the XML is referenced in the [docs](https://docs.camunda.io/docs/reference/bpmn-workflows/timer-events/timer-events#additional-resources).

**Comment by mumukiller** [comment]
Just to confirm - it just accepts only a constant, not a variable, right?
I'm asking because i got this
 ERROR: Invalid timer date expression (Invalid date-time format '"= someVariable"' for expression '"= someVariable"')

**Comment by saig0** [comment]
> Just to confirm - it just accepts only a constant, not a variable, right?

No. It worked for an intermediate timer event with an expression (on Zeebe version `0.26.0`).

```
    <bpmn:intermediateCatchEvent id="Event_1y1ox1z">
      <bpmn:incoming>Flow_0xxbekp</bpmn:incoming>
      <bpmn:outgoing>Flow_0lqu3lu</bpmn:outgoing>
      <bpmn:timerEventDefinition id="TimerEventDefinition_12246mi">
        <bpmn:timeDate xsi:type="bpmn:tFormalExpression">= now() + duration("PT10S")</bpmn:timeDate>
      </bpmn:timerEventDefinition>
    </bpmn:intermediateCatchEvent>
```

_However, I can not guarantee anything since it is not supported officially yet._

**Comment by stephanpelikan** [comment]
As I'm migrating C7 BPMNs to C8 I hit also this issue. For typical business processes absolute dates are really important:

1. Most of the time one has to deal with business days, so the date-time is calculated.
2. If only durations are available for calculating this value one needs to
    1. Prepare this in a separate (service-)task immediately before the timer event = modelling technical tasks instead of meaningful tasks
    2. Prepare this in an existing (service-)task immediately before the timer event = abuse of a semantical independent task
    3. Prepare this in an upfront section of the process, if possible = this might lead to a bigger delay between desired point in time and the calculated depending on the execution of the process until the timer event gets active
    4. As a workaround I used this duration expression: `= date and time(offeringDeadline) - now()` which is the most precise point in time but it is verbose.

Since with Zeebe one has to bring all data required by the BPMN into Zeebe as variables using relative values should be avoided as you always have to set those values in tasks upfront and there is no quarantee that timers get activated as expected (slow completion of intermediate tasks, incidents, etc.). I'm sure you are pretty aware of all this mentioned, so please take this overall comment as a vote for this feature ;-)

**Comment by nikku** [comment]
Expected to work by a user [here](https://github.com/camunda/camunda-modeler/issues/3516).

**Comment by aleksander-dytko** [comment]
Hi @asaf @themichaelscott @stephanpelikan @mumukiller as we have released it with Camunda Platform 8.3-alpha2 (released June) and Camunda Modeler 5.12 I'm wondering if everything works for you as expected. Could you provide your feedback?
Thanks!
